tests/TestCase.php

```diff
@@ -2,9 +2,52 @@
 
 namespace Tests;
 
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\MorphMany;
+use Illuminate\Database\Eloquent\Relations\MorphTo;
 use Illuminate\Foundation\Testing\TestCase as BaseTestCase;
 
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function hasMany_relation_test($related, $foreignKey, $relation)
+    {
+        $this->assertInstanceOf(HasMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function belongsTo_relation_test($related, $foreignKey, $ownerKey, $relation)
+    {
```

Comment:
giải thích các function test trong này

Reply:
Các function được dùng để test các relation hasMany, belongsTo, belongsToMany, morphMany, mophTo.
Trong function hasMany_relation_test:
$this->assertInstanceOf(HasMany::class, $relation); để test xem $relation có phải là loại HasMany không.
$this->assertInstanceOf($related, $relation->getRelated()); để test quan hệ model tương ứng.
$this->assertEquals($foreignKey, $relation->getForeignKeyName()); để test khóa ngoại